07.bowling_object/lib/game.rb

```diff
@@ -12,28 +12,24 @@ def initialize(marks)
 
   def score
     frames = generate_frames
-    score = 0
-    frames.each_with_index do |frame, idx|
-      score += idx == 9 ? frame.total_score : score_nonfinal(frame, frames, idx)
+    frames.each_with_index.sum do |frame, idx|
+      idx == 9 ? frame.total_score : score_nonfinal(frame, frames, idx)
     end
-    score
   end
 
   private
 
   def generate_frames
-    rolls = parse_marks
-    rolls.map { |f| Frame.new(*f[0..2]) }
+    rolls = generate_rolls
+    rolls.map { |roll| Frame.new(*roll[0..2]) }
```

Comment:
```suggestion
    generate_rolls.map { |roll| Frame.new(*roll) }
```

メソッドチェーンにしてもよさそうです。
あと`[0..2]`なしでも動いたりしませんか？